File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var uniqueOccurrences = function(arr) {
  const map = new Map()
  arr.forEach(n => map.set(n, map.has(n) ? map.get(n) + 1 : 1))
  const counts = [...map.values()]
  return counts.every((c, i) => counts.indexOf(c) === i)
};

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.go
func uniqueOccurrences(arr []int) bool {
  n := len(arr)
  counter := make(map[int]int, n)
  for _, v := range arr {
    counter[v]++
  }
  cnt2 := make(map[int]int, n)
  for _, v := range counter {
    cnt2[v]++
    if cnt2[v] > 1 {
      return false
    }
  }
  return true
}